src/sentry/utils/email.py

```diff
@@ -109,7 +114,15 @@ def make_msgid(domain):
     return msgid
 
 
-FROM_EMAIL_DOMAIN = domain_from_email(settings.DEFAULT_FROM_EMAIL)
+# cache the domain_from_email calculation
+_from_email_domain_cache = [None, None]
+
+
+def get_from_email_domain():
+    from_ = options.get('mail.from')
+    if not _from_email_domain_cache[0] == from_:
+        _from_email_domain_cache[0] = domain_from_email(from_)
+    return _from_email_domain_cache[1]
```

Comment:
I don't understand this function.

1. Won't unless `domain_from_email(from_) == from_`, won't this conditional block always get executed?
2. If `_from_email_domain_cache[0]` is being tested and set, why isn't it used?
3. When is the second item of the `_from_email_domain_cache` list getting set?

Is L124 supposed to be `_from_email_domain_cache[1] = …`?